### PR TITLE
Fixes #16661: inneficient computation of RuleStatusReports and NodeStatusReports

### PR DESCRIPTION
**Pull request by ncharles**
https://issues.rudder.io/issues/16661

### CODE REVIEW

**Comment by ncharles** [comment]
this is a WIP. I'll ensure that getNodeStatusReports return merged data, and don't use merge in all method calling it

**Comment by ncharles** [comment]
PR updated with a new commit

**Comment by ncharles** [comment]
PR updated with a new commit

**Comment by ncharles** [comment]
PR updated with a new commit

**Comment by ncharles** [comment]
PR updated with a new commit

**Comment by ncharles** [comment]
PR updated with a new commit

**Comment by ncharles** [comment]
PR updated with a new commit

**Comment by ncharles** [comment]
PR updated with a new commit

**Comment by ncharles** [comment]
PR updated with a new commit

**Comment by ncharles** [comment]
PR updated with a new commit

**Comment by ncharles** [comment]
PR updated with a new commit

**Comment by ncharles** [comment]
PR updated with a new commit

**Comment by ncharles** [comment]
This version caused a FullGC during policy compliance computation + writting policies
It never happened without this PR

**Comment by ncharles** [comment]
PR updated with a new commit

**Comment by ncharles** [comment]
PR updated with a new commit

**Comment by ncharles** [comment]
ping @fanf

File: webapp/sources/rudder/rudder-web/src/main/scala/com/normation/rudder/web/snippet/HomePage.scala [hunk 1]
```diff
@@ -184,7 +184,19 @@ class HomePage extends Loggable {
...
+      global   = if(reports.isEmpty) {
+                             None
+                           } else {
+                             val complianceLevel = ComplianceLevel.sum(compliancePerNodes.map(_._2))
+                             Some((
+                               complianceLevel
+                             , complianceLevel.complianceWithoutPending.round
+                             ))
+                           }
+   //   global    =  reportingService.computeComplianceFromReports(reports)
```
Comment (fanf) [inline webapp/sources/rudder/rudder-web/src/main/scala/com/normation/rudder/web/snippet/HomePage.scala]: remove comments here and on l208

**Comment by ncharles** [comment]
PR updated with a new commit

**Comment by ncharles** [comment]
OK, squash merging this PR

**Comment by ncharles** [comment]
OK, merging this PR

**Pull request merged**